src/Entity/Access/ApiDocAccessControlHandler.php

```diff
@@ -59,6 +98,60 @@ protected function checkAccess(EntityInterface $entity, $operation, AccountInter
     return $parent_access;
   }
 
+  /**
+   * Additional access control for revisions.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity for which to check access.
+   * @param string $operation
+   *   The entity operation.
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   The user for which to check access.
+   *
+   * @return \Drupal\Core\Access\AccessResultInterface
+   *   The access result.
+   */
+  protected function checkAccessRevisions(EntityInterface $entity, $operation, AccountInterface $account) {
+    /** @var \Drupal\Core\Entity\EntityStorageInterface $entity_storage */
+    $entity_storage = $this->entityTypeManager->getStorage($this->entityTypeId);
+
+    // Must have access to the same operation on the default revision.
+    $default_revision = $entity_storage->load($entity->id());
+    $has_default_entity_rev_access = $default_revision->access($operation, $account);
+    if (!$has_default_entity_rev_access) {
+      return AccessResult::forbidden();
+    }
+
+    $map = [
+      'view' => "view all {$this->entityTypeId} revisions",
+      'update' => "revert all {$this->entityTypeId} revisions",
+      'delete' => "delete all {$this->entityTypeId} revisions",
+    ];
+    $bundle = $entity->bundle();
```

Comment:
There aren't any bundles or a way to create bundles and the `view apidoc apidoc revisions` permission does not exist.

Reply:
I removed the permission for now.